tests/rest/test_endpoints.py

```diff
@@ -109,27 +123,73 @@ async def test_model_docs(rest_client, model_name: str, model_version: Optional[
     "model_name,model_version", [("sum-model", "v1.2.3"), ("sum-model", None)]
 )
 async def test_infer(
-    rest_client,
-    inference_request,
-    model_name,
-    model_version,
+    rest_client: AsyncClient,
+    inference_request: InferenceRequest,
+    model_name: str,
+    model_version: Optional[str],
 ):
     endpoint = f"/v2/models/{model_name}/infer"
     if model_version is not None:
         endpoint = f"/v2/models/{model_name}/versions/{model_version}/infer"
+<<<<<<< HEAD
     response = await rest_client.post(endpoint, json=inference_request.model_dump())
+=======
+>>>>>>> Included endpoint tests for generate and generate_stream.
 
+    response = await rest_client.post(endpoint, json=inference_request.dict())
     assert response.status_code == 200
 
     prediction = InferenceResponse.model_validate(response.json())
     assert len(prediction.outputs) == 1
     assert prediction.outputs[0].data == TensorData(root=[6])
 
 
+@pytest.mark.parametrize("sum_model", [lazy_fixture("text_model")])
+@pytest.mark.parametrize(
+    "model_name,model_version", [("text-model", "v1.2.3"), ("text-model", None)]
+)
+async def test_generate(
```

Comment:
can `generate` test be an extra parametrized item in `infer` test?

Reply:
It is a bit tricky to parameterise the model loading through lazy_fixtures due to recursive dependency involving fixture. I will leave it like this since I don't want to refactor the tests.